test-clients/src/main/java/com/hedera/services/bdd/suites/reconnect/ValidateChangesAfterReconnect.java

```diff
@@ -0,0 +1,158 @@
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static com.hedera.services.bdd.spec.HapiApiSpec.customHapiSpec;
+import static com.hedera.services.bdd.spec.assertions.TransactionRecordAsserts.recordWith;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getAccountBalance;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getAccountInfo;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getFileContents;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getTxnRecord;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.cryptoCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.fileCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.fileUpdate;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.makeFree;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.sleepFor;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withLiveNode;
+import static com.hederahashgraph.api.proto.java.HederaFunctionality.CryptoGetInfo;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.AUTORENEW_DURATION_NOT_IN_RANGE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.DUPLICATE_TRANSACTION;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.NOT_SUPPORTED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+
+public class ValidateChangesAfterReconnect extends HapiApiSuite {
+
+	private static final Logger log = LogManager.getLogger(ValidateChangesAfterReconnect.class);
+
+	final String PATH_TO_VALID_BYTECODE = HapiSpecSetup.getDefaultInstance().defaultContractPath();
+
+	public static void main(String... args) {
+		new ValidateAppPropertiesStateAfterReconnect().runSuiteSync();
+	}
+
+	@Override
+	protected List<HapiApiSpec> getSpecsInSuite() {
+		return List.of(
+				validateChangesAfterReconnect()
+		);
+	}
+
+	private HapiApiSpec validateChangesAfterReconnect() {
+		final String fileInfoRegistry = "apiPermissionsReconnect";
+		final String transactionId = "specialTransactionId";
+		final String transactionFeeid = "authorizedTxn";
+		final long newFee = 159_588_904;
+
+		return customHapiSpec("validateChangesAfterReconnect")
+				.withProperties(Map.of(
+						"txn.start.offset.secs", "-5")
+				)
+				.given(
+						sleepFor(Duration.ofSeconds(25).toMillis()),
+						getAccountBalance(GENESIS)
+								.setNode("0.0.6")
+								.unavailableNode()
+				)
+				.when(
+						getAccountBalance(GENESIS)
+								.setNode("0.0.6")
+								.unavailableNode(),
+
+						fileUpdate(APP_PROPERTIES)
+								.payingWith(ADDRESS_BOOK_CONTROL)
+								.overridingProps(Map.of("minimumAutoRenewDuration", "20")),
+
+						fileCreate("effectivelyImmutable")
+								.contents("Can't touch me!"),
```

Comment:
The file name and contents might give a misleading understanding. The file is actually not updatable, due to the updated API permissions, not due to itself (ACL for example). It is still deletable though...
I would suggest we move the API permissions update to go in front of it, then provide a file name and contents to describe that the file is not updatable because of this API permissions update.